src/app/core/vehicles/vehicles-rest.service.ts

```diff
@@ -0,0 +1,22 @@
+import { HttpClient } from '@angular/common/http';
+import { VehiclesService } from "./vehicles.service";
+import { Injectable } from "@angular/core";
+import { Vehicle } from "../../shared/models/vehicle";
+import { environment } from './../../../environments/environment';
+import { Observable } from 'rxjs';
+
+
+@Injectable()
+export class VehiclesRestService implements VehiclesService {
+
+  public constructor(private http: HttpClient) { }
+  
+  public getVehicles(): Observable<Vehicle[]> {
+    return this.http.get<Vehicle[]>(environment.api + '/vehicles');
+  }
+
+  public addVehicle(vehicle: Vehicle): Observable<Vehicle> {
+    return this.http.post<Vehicle>(environment.api + '/vehicles', vehicle);
```

Comment:
Przemapuj sobie Vehicle na VehicleDTO/VehicleRequest/NewVehicleRequest/Whatever. Gdy zmieni się REST API to będziesz musiał zmienić sobie logikę mapowania, bez zmiany twojego modelu.